components/style/parser.rs

```diff
@@ -81,3 +81,8 @@ impl<'a> ParserContext<'a> {
 pub fn log_css_error(input: &mut Parser, position: SourcePosition, message: &str, parsercontext: &ParserContext) {
     parsercontext.error_reporter.report_error(input, position, message);
 }
+
+/// XXXManishearth use this for everything
```

Comment:
The advantage is just consistency 😄 

There's a slight hitch here -- some type parse with a context, some don't. I'd need two traits that provide `parse()` and `context_parse()` to implement it everywhere. Which I'd rather do as a followup.